bittensor/_neuron/text/core_server/run.py

```diff
@@ -160,44 +160,38 @@ def registration_check():
 
         # Check for stake
         def stake_check() -> bool:
-                
             # Check stake.
             uid = metagraph.hotkeys.index(pubkey)
             if metagraph.S[uid].item() < config.neuron.blacklist.stake:
                 raise Exception('Stake blacklist')
             return False
 
-        def validator_check():
-
-            uid = metagraph.hotkeys.index(pubkey)
-            if (metagraph.W[uid] >0).sum() >= n_topk_peer_weights:
-                return False
-
-            raise Exception('Validator blacklist')
-
-
         # Check for time
         def time_check():
             current_time = datetime.now()
-            if pubkey in timecheck.keys():
-                prev_time = timecheck[pubkey]
-                if current_time - prev_time >= timedelta(seconds=config.neuron.blacklist.time):
-                    timecheck[pubkey] = current_time
-                    return False
+            # Only check if the request are forward requests
+            if request_type == 'FORWARD':
+                if pubkey in timecheck.keys():
+                    prev_time = timecheck[pubkey]
+                    if current_time - prev_time >= timedelta(seconds=config.neuron.blacklist.time):
```

Comment:
Regarding the time check, 
(1) could we again take a moving average to sample across multiple requests from the same validator, so that it would be a more fair estimation?
(2) maybe it is better to correlate it with stake? ie. if you have higher stake, then the server would allow more frequency. But higher stake =/= more weight set to the server, sooooo it might not be a good metric.

Reply:
(1) haha wait I'm confused, a moving average of the time between requests? So we reject requests if the moving average drops below a certain value?

Reply:
(2) Yea, I think it might be better to separate the two. Blacklist is designed primarily for protection rather than the priority of service.